src/typechecker.erl

```diff
@@ -4078,6 +4078,11 @@ handle_type_error({undef, user_type, Anno, {Name, Arity}}, Opts) ->
               [format_location(Anno, brief, Opts),
                Name, Arity,
                format_location(Anno, verbose, Opts)]);
+handle_type_error({undef, user_type, Anno, {Module, Name, Arity}}, Opts) ->
```

Comment:
The code which was supposed to handle this type error is on line 4070, the clause with the pattern `{undef, Type, {{atom, _, Module}, {atom, _, Name}, Arity}}`. That clause used to match in some earlier OTP version, where the module and function name were represented as `{atom, _, _}`s.

So, please remove the duplicate one and preferable add a test case so we can see if it works in all our supported OTP versions.

Reply:
I tried to create a test but any test I could come up with is stopped by the compiler. I think these cases can#t ever happen so I#m removing them.

Reply:
But it happened, right?

I think it can't happen for beam files, but for erl files it can, because of the compiler checks that come after parsing.

I'm pretty sure gradualizer_db can return not_found in some case. If it can't, we should remove that code too.